internal/indexer/postgres/setlayerscanned.go

```diff
@@ -2,51 +2,49 @@ package postgres
 
 import (
 	"context"
+	"errors"
 	"fmt"
 
 	"github.com/jackc/pgx/v4"
-	"github.com/jackc/pgx/v4/pgxpool"
+
 	"github.com/quay/claircore"
 	"github.com/quay/claircore/internal/indexer"
 )
 
-func setLayerScanned(ctx context.Context, pool *pgxpool.Pool, hash claircore.Digest, scnr indexer.VersionedScanner) error {
-	const (
-		selectScanner = `
-		SELECT id
-		FROM scanner
-		WHERE name = $1
-		  AND version = $2
-		  AND kind = $3;
-		`
-		query = `
-		WITH layers AS (
-			SELECT id AS layer_id
-			FROM layer
-			WHERE hash = $1
+func (s *store) SetLayerScanned(ctx context.Context, hash claircore.Digest, scnr indexer.VersionedScanner) error {
+	// TODO(hank) This could be reworked into a single query.
+	const query = `
+WITH
+	layers
+		AS (
+			SELECT id AS layer_id FROM layer WHERE hash = $1
 		)
-		INSERT
-		INTO scanned_layer (layer_id, scanner_id)
-		VALUES ((SELECT layer_id FROM layers),
-				$2)
-		ON CONFLICT (layer_id, scanner_id) DO NOTHING;
+INSERT
+INTO
+	scanned_layer (layer_id, scanner_id)
+VALUES
+	((SELECT layer_id FROM layers), $2)
+ON CONFLICT
+	(layer_id, scanner_id)
+DO
+	NOTHING;
 		`
-	)
 
 	// get scanner id
-	row := pool.QueryRow(ctx, selectScanner, scnr.Name(), scnr.Version(), scnr.Kind())
+	row := s.pool.QueryRow(ctx, selectScanner, scnr.Name(), scnr.Version(), scnr.Kind())
```

Comment:
do you want to replace this block with the store.selectScanners method?

Reply:
I meant to rework this because it's doing something odd and only selecting one scanner, so I'll do that.